client/views/app/room.coffee

```diff
@@ -188,7 +188,9 @@ Template.room.helpers
 		return !! ChatRoom.findOne { _id: @_id, t: 'c' }
 
 	canRecordAudio: ->
-		return RocketChat.settings.get('Message_AudioRecorderEnabled') and (navigator.getUserMedia? or navigator.webkitGetUserMedia?)
+		wavRegex = /audio\/wav|audio\/\*/i
+		wavEnabled = RocketChat.settings.get("FileUpload_MediaTypeWhiteList").match(wavRegex)
+		return RocketChat.settings.get('Message_AudioRecorderEnabled') and (navigator.getUserMedia? or navigator.webkitGetUserMedia?) and wavEnabled
 
```

Comment:
I would also check if [file uploads are enabled](https://github.com/RocketChat/Rocket.Chat/blob/master/packages/rocketchat-lib/settings/server/startup.coffee#L44). IMHO this rule should have higher precedence over others, and we should not allow audio records if uploads are disabled.